imgw_pipeline.py

```diff
@@ -20,13 +49,95 @@ def main(start_year: int, end_year: int) -> None:
     end_year : int
         The end year of the range (inclusive) for validating ZIP files.
+    download_data : bool
+        Boolean indicating whether to download and unzip the data or
+        operate on the default data directory.
 
     Returns
     -------
     None
     """
-    obtained_files = obtain_data(start_year, end_year)
-    print(obtained_files)
+    start_time = datetime.now()
+    if download_data:
+        obtain_data(start_year, end_year)
+        print(f"Download and unzip data: DONE (1/{NUMBER_OF_TASKS})")
+    else:
+        print(f"Download and unzip data: SKIPPED (1/{NUMBER_OF_TASKS})")
+
+    merged_df = merge_unzipped_data(
+        "data/interim", start_year, end_year, **IMGW_PRECIP_DATA_READ_OPTIONS
+    )
+    print(f"Merge unzipped data: DONE (2/{NUMBER_OF_TASKS})")
+
+    get_missing_stations_info(merged_df, **IMGW_STATIONS_INFO_READ_OPTIONS)
+    print(f"Generate missing stations report: DONE (3/{NUMBER_OF_TASKS})")
+
+    df_with_stations_info = append_stations_info(
+        merged_df, **IMGW_STATIONS_INFO_READ_OPTIONS
+    )
+    print(f"Add stations info: DONE (4/{NUMBER_OF_TASKS})")
+
+    df_with_stations_info = transform_coords(
+        df_with_stations_info, ["lat", "lon"]
+    )
+    print(f"Transform coords: DONE (5/{NUMBER_OF_TASKS})")
+
+    plot_stations(df_with_stations_info)
+    print(f"Generate stations plot: DONE (6/{NUMBER_OF_TASKS})")
+
+    voivodeships_mapping = create_stations_voivodeship_mapping(
+        df_with_stations_info
+    )
+    print(f"Generate stations voivodeship mapping: DONE (7/{NUMBER_OF_TASKS})")
+
+    analyzed_subset = get_chosen_voivodeship_stations(
+        df_with_stations_info, voivodeships_mapping, CHOSEN_VOIVODESHIP
+    )
+    print(f"Get chosen voivodeship stations: DONE (8/{NUMBER_OF_TASKS})")
+
+    analyzed_subset_tst = convert_date_info_to_timestamp(
+        analyzed_subset, ["year", "month", "day"]
+    )
+    print(f"Convert date to timestamp: DONE (9/{NUMBER_OF_TASKS})")
+
+    analyzed_subset_subset = drop_stations_below_threshold(
+        analyzed_subset_tst, PERCENT_OF_DATES_THRESHOLD
+    )
+    print(f"Drop stations with too little data: DONE (10/{NUMBER_OF_TASKS})")
+
+    full_dates = fill_missing_dates(analyzed_subset_subset)
+    print(f"Fill missing dates: DONE (11/{NUMBER_OF_TASKS})")
+
+    interpolated = interpolate_total_precip(full_dates)
+    print(f"Interpolate missing dates values: DONE (12/{NUMBER_OF_TASKS})")
+
+    output_files = []
+    for idx, window in enumerate(WINDOWS_TO_CALCULATE):
+        rolling_precip_sum = calculate_stations_rolling_precipitation_sum(
+            interpolated, window
+        )
+        output_filename = os.path.join(OUTPUTS_PATH, f"spi_{window}.csv")
+        spi = calculate_stations_spi(
+            rolling_precip_sum,
+            save_to_csv=True,
+            output_csv_path=output_filename,
+        )
+        output_files.append(output_filename)
+        plot_spi_and_total_precip_time_series(
```

Comment:
You can see some suspicious outliers in the data on the charts, is your next step to investigate that?

Reply:
I will try to investigate if I have enough time left :) If not, I will try to include the possible causes of the outliers in the report